src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/TrieRangeIterator.java

```diff
@@ -0,0 +1,268 @@
+                            payloadedNode = position;
+                    }
+                    else
+                        payloadedNode = -1;
+
+                limitByte = 256;
+                if (atLimit)
+                {
+                    limitByte = limit.next();
+                    if (s < limitByte)
+                        atLimit = false;
+                }
+                if (childIndex < 0)
+                    break;
+
+                int transitionByte = transitionByte(childIndex);
+                collector.add(transitionByte);
+
+                prev = new IterationPosition(position, childIndex, limitByte, prev);
+                go(transition(childIndex));
+            }
+
+            childIndex = -1 - childIndex - 1;
+
+            stack = new IterationPosition(position, childIndex, limitByte, prev);
+
+            // Advancing now gives us first match if we didn't find one already.
+            if (payloadedNode != -1)
+                next = payloadedNode;
+            else {
+                next = advanceNode();
+            }
+        }
+        catch (Throwable t)
+        {
+            super.close();
+            throw t;
+        }
+    }
+
+    private long advanceNode()
+    {
+        long child;
+        int transitionByte;
+
+        go(stack.node); // can throw NotInCacheException, OK no state modified yet
+
+        while (true)
+        {
+            // advance position in node but don't change the stack just yet due to NotInCacheExceptions
+            int childIndex = stack.childIndex + 1;
+            transitionByte = transitionByte(childIndex);
+
+            if (transitionByte > stack.limit)
+            {
+                // ascend
+                stack = stack.prev;
+                if (stack == null)        // exhausted whole trie
+                    return -1;
+                collector.pop();
+                go(stack.node); // can throw NotInCacheException, OK - stack ready to re-enter loop with parent
+                continue;
+            }
+
+            child = transition(childIndex);
+
+            if (child != -1)
+            {
+                assert child >= 0 : String.format("Expected value >= 0 but got %d - %s", child, this);
+
+                // descend
+                go(child); // can throw NotInCacheException, OK - stack not yet changed, limit not yet incremented
+
+                int l = 256;
+                if (transitionByte == stack.limit)
+                    l = limit.next();
+
+                stack.childIndex = childIndex;
+                stack = new IterationPosition(child, -1, l, stack);
+
+                collector.add(transitionByte);
+
+                if (payloadFlags() != 0 && (inclEnd || l >= 0))
+                    return child;
+            }
+            else
+            {
+                stack.childIndex = childIndex;
+            }
+        }
+    }
+
+    private void initializeNoLeftBound(long root, int limitByte)
```

Comment:
nit: Looks like this is identical to the `initializeNoLeftBound()` in `ValueIterator`.

Reply:
As above.